src/app/pages/settings.cpp

```diff
@@ -341,6 +349,65 @@ QWidget *LayoutSettingsTab::control_bar_widget()
     return widget;
 }
 
+QWidget *LayoutSettingsTab::fullscreen_mode_widget()
+{
+    QWidget *widget = new QWidget(this);
+    QHBoxLayout *layout = new QHBoxLayout(widget);
+
+    QLabel *label = new QLabel("Fullscreen mode", widget);
+    layout->addWidget(label, 1);
+
+    Switch *toggle = new Switch(widget);
+    toggle->scale(this->arbiter.layout().scale);
+    toggle->setChecked(this->arbiter.layout().fullscreen_mode);
+    connect(toggle, &Switch::stateChanged, [this](bool state){ this->arbiter.set_fullscreen_mode(state); });
```

Comment:
youll also want to update the switch state if a user double clicks to go fullscreen (to test double click when on settings/layout)